spec/unit/provider/package/zypper_spec.rb

```diff
@@ -136,7 +127,7 @@ def shell_out_expectation!(command, options=nil)
       shell_out_expectation!(
         "zypper --non-interactive install --auto-agree-with-licenses emacs=1.0"
```

Comment:
i think we get the correct behavior through the idempotency checks in the parent provider.   if the package is already installed then we won't call either install or upgrade on it if the action is install.  if the action is upgrade then we'll issue the command.  i'd argue 'zypper install' has broken behavior, but it doesn't actually matter to us...